programmers/난이도별/level01.서울에서_김서방_찾기/HyeonJeong.java

```diff
@@ -0,0 +1,15 @@
+package level01.서울에서_김서방_찾기;
+
+public class HyeonJeong {
+    public String solution(String[] seoul) {
+        String answer = "";
+        for (int i = 0; i < seoul.length; i++)
+            if (seoul[i] == "Kim")
+                answer = "김서방은 " + i + "에 있다";
+            return answer;
```

Comment:
```suggestion
        for (int i = 0, len = seoul.length; i < len; i++)
            if (seoul[i] == "Kim")
                return "김서방은 " + i + "에 있다";
```

- 배열 혹은 문자열의 길이를 반복 할 때 마다 검사 하는 것은 비효율적이므로 변수로 빼주세요!
- answer 변수를 선언하고 할당할 필요 없이 if문 안에서 바로 return 하면 될 것 같습니다.

그리고 python으로 풀이한 것 처럼 똑같이 할 수 있어요!

```java
    return "김서방은 " + seoul.indexOf("Kim") + "에 있다";
```

참고 : https://mine-it-record.tistory.com/124

Reply:
return을 if문 안에 사용하면 return이 누락되었다고 뜹니다.
그리고 프로그래머스에서는 위 코드로 고친 것이 두가지 다 안 돌아가네요...

Reply:
앗.. 제가 문제 이해를 잘못하고 있었네요 ㅋㅋ
다음과 같이 풀이할 수 있습니다.

```java
    public String solution(String[] seoul) {
        return String.format("김서방은 %d에 있다", Arrays.asList(seoul).indexOf("Kim"));
    }
```